services/tracing/tracing/infra/store_psql.go

```diff
@@ -103,13 +104,13 @@ func (s *stepStore) QueryTraces(filter tracing.Filter, r pagination.Request) (*p
 func (s *stepStore) GetStepsByTracingIDs(tracingIds []string) ([]tracing.EnrichedStep, error) {
 	q := sq.Select(
 		"tracing_id",
-		"device_id",
+		"COALESCE(device_id, 0)",
```

Comment:
what does coalesce here do?

Reply:
If device_id is NULL it instead returns 0. Allowing the trace service to just `lo.Max(tace.steps.deviceID)` to get the device ID. As well as saving some null pointer checks.